### PR TITLE
Remove mps workaround for fp16 GELU, which is now supported natively

### PR DESCRIPTION
**Pull request by skotapati**
# What does this PR do?

Fixes MPS fp16 GELU calls being forced to cast to fp32. fp16 GELU is now supported natively in MPS so this workaround is no longer needed

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by pcuenca** [comment]
Thanks a lot @skotapati! Do you know the minimum PyTorch version that benefits from this?

**Comment by hlky** [comment]
@pcuenca Looks like it's been fixed since 1.13 https://github.com/pytorch/pytorch/pull/86218 the code was originally added https://github.com/huggingface/diffusers/pull/1533 and at that time the minimum supported version for PyTorch was 1.4

**Comment by pcuenca** [comment]
I tested on `1.13.1` and it still failed, but it worked on `2.0.0`. I'd suggest we add a condition to check for a minimum version of PyTorch >= 2.

**Comment by skotapati** [comment]
Makes sense, adding a check for torch>2.0

**Comment by skotapati** [comment]
@pcuenca thanks for the review, added the fallback for torch < 2.0

File: src/diffusers/models/activations.py [hunk 1]
```diff
@@ -79,10 +79,10 @@ def __init__(self, dim_in: int, dim_out: int, approximate: str = "none", bias: b
         self.approximate = approximate
 
     def gelu(self, gate: torch.Tensor) -> torch.Tensor:
-        if gate.device.type != "mps":
-            return F.gelu(gate, approximate=self.approximate)
-        # mps: gelu is not implemented for float16
-        return F.gelu(gate.to(dtype=torch.float32), approximate=self.approximate).to(dtype=gate.dtype)
+        if gate.device.type == "mps" and torch.__version__ < '2.0.0':
```
Comment (yiyixuxu) [inline src/diffusers/models/activations.py:82]: we have a is_torch_version function that you can use here https://github.com/huggingface/diffusers/blob/fa3a9100bede6d45c84a68f10cef45a7c562ac94/src/diffusers/utils/import_utils.py#L690

**Review by pcuenca — APPROVED** [review]

**Comment by yiyixuxu** [comment]
thanks @skotapati @pcuenca !

**Pull request merged**